script.js

```diff
@@ -0,0 +1,165 @@
+const toDoInput = toDoForm.querySelector("input");
+const toDoInputBtn = document.querySelector(".todo-input").querySelector("div");
+const toDoListPre = document.querySelector(".todo-list-pre");
+const toDoListDone = document.querySelector(".todo-list-done");
+
+let toDoList = [];
+let indPre = 0;
+let indDone = 0;
+
+const createToDo = (event) => {
+  event.preventDefault();
+
+  const toDoObj = {
+    done: false,
+    text: toDoInput.value,
+    id: indPre + 1,
+  };
+
+  drawToDo(toDoObj);
+  toDoInput.value = "";
+  indPre += 1;
+
+  saveListCache();
+};
+
+const drawToDo = (toDoObj) => {
+  const li = document.createElement("li");
+
+  const span = document.createElement("span");
+  span.addEventListener("click", toggleToDo);
+  span.innerHTML = toDoObj.text;
+  li.appendChild(span);
+
+  const delButton = document.createElement("img");
+  delButton.src = "img/ios_trash.png";
+  delButton.addEventListener("click", deleteToDo);
+  li.appendChild(delButton);
+
+  li.id = toDoObj.id;
+
+  if (toDoObj.done === false) {
+    toDoListPre.appendChild(li);
+    document.getElementById("pre").innerHTML =
+      Number(document.getElementById("pre").innerHTML) + 1;
+  } else {
+    toDoListDone.appendChild(li);
+    document.getElementById("done").innerHTML =
+      Number(document.getElementById("pre").innerHTML) + 1;
+  }
+};
+
+const deleteToDo = (event) => {
+  const { target } = event;
+  const li = target.parentNode;
+  const done = li.parentNode.className == "todo-list-done";
+
+  if (!done) {
+    toDoListPre.removeChild(li);
+    document.getElementById("pre").innerHTML =
+      Number(document.getElementById("pre").innerHTML) - 1;
+  } else {
+    toDoListDone.removeChild(li);
+    document.getElementById("done").innerHTML =
+      Number(document.getElementById("pre").innerHTML) - 1;
+  }
+
+  saveListCache();
+};
+
+const toggleToDo = (event) => {
+  const { target } = event;
+  const li = target.parentNode;
+  const done = li.parentNode.className == "todo-list-done";
+
+  if (!done) {
+    toDoListPre.removeChild(li);
+    toDoListDone.appendChild(li);
+    document.getElementById("pre").innerHTML =
+      Number(document.getElementById("pre").innerHTML) - 1;
+    document.getElementById("done").innerHTML =
+      Number(document.getElementById("done").innerHTML) + 1;
+    indDone += 1;
+  } else {
+    toDoListDone.removeChild(li);
+    toDoListPre.appendChild(li);
+    document.getElementById("pre").innerHTML =
+      Number(document.getElementById("pre").innerHTML) + 1;
+    document.getElementById("done").innerHTML =
+      Number(document.getElementById("done").innerHTML) - 1;
+    indPre += 1;
+  }
+
+  saveListCache();
+};
+
+const saveListCache = () => {
+  saveList = [];
+
+  saveList.push(
+    Number(document.querySelectorAll(".todo-list-pre > li > span").length)
+  );
+  for (let toDoObj of document.querySelectorAll(".todo-list-pre > li > span")) {
+    saveList.push(toDoObj["innerHTML"]);
+  }
```

Comment:
es6 문법을 최대한 활용하는게 좋겠죠. 다만 `querySelectorAll`의 리턴값 같이 `NodeList` 타입의 객체는 `forEach`같은 array method를 사용할 수 없으니 이렇게 하면 됩니다.

```js
Array.from(document.querySelectorAll(".todo-list-pre > li > span"))
  .forEach((el) => 
    saveList = [...saveList, toDoObj['innerHTML']];
  )
```

Reply:
또한 `for-of` 문법을 사용하실 때에 iterator는 `for`문 context 외부에 미리 선언해주는게 좋습니다. 이런 방식의 경우 반복문의 매 실행마다 `toDoObj`를 매번 재선언하게 됩니다.
```suggestion
  let toDoObj;
  
  for (toDoObj of document.querySelectorAll(".todo-list-pre > li > span")) {
    saveList.push(toDoObj["innerHTML"]);
  }
```

Reply:
최근에 항상 갖고 있던 생각이, 제가 너무 예전 자바스크립트 문법들만 기억하고 있는 것 같아
ES6 문법들을 숙지하고 좀 더 모던 자바스크립트에 가까운 코드를 작성하고 싶었는데
이번에 선종님께서 주신 리뷰들이 너무 큰 가르침이 될 것 같습니다. 😌